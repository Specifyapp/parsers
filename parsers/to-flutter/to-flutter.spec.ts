import seeds from '../../tests/seeds';
import toFlutter from './to-flutter.parser';

describe('To Flutter', () => {
  it('Should generate list of files containing design tokens', async () => {
    const tokens = seeds().tokens;
    const result = await toFlutter(tokens, undefined);
    expect(Array.isArray(result)).toBeTruthy();
    expect(result).toMatchSnapshot();
  });

  it('Should generate the color theme with custom file name and custom class name', async () => {
    const tokens = seeds().tokens;
    const result = await toFlutter(
      tokens.filter(({ type }) => type === 'color'),
      {
        formatByType: {
          color: {
            fileName: 'custom-colors-file-name.dart',
            className: 'LightTheme',
          },
        },
      },
    );
    expect(Array.isArray(result)).toBeTruthy();
    expect(result).toMatchSnapshot();
  });

  it('Should generate the measurement theme with custom file name, class name and ratio', async () => {
    const tokens = seeds().tokens;
    const result = await toFlutter(
      tokens.filter(({ type }) => type),
      {
        formatByType: {
          measurement: {
            fileName: 'custom-measurements-file-name.dart',
            className: 'CustomTheme',
            devicePixelRatio: 3.0,
          },
        },
      },
    );
    expect(Array.isArray(result)).toBeTruthy();
    expect(result).toMatchSnapshot();
  });
});
