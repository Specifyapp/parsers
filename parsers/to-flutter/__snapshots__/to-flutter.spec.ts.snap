// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`To Flutter Should generate list of files containing design tokens 1`] = `
Array [
  Object {
    "name": "measurements.dart",
    "value": Object {
      "content": "
import 'dart:ui';

class SpecifyMeasurement {
    SpecifyMeasurement._();

    static const baseSpace01 = 8.00;
    static const baseSpace02 = 16.00;
    static const baseSpace03 = 24.00;
    static const baseSpace04 = 32.00;
    static const baseSpace05 = 64.00;
    static const baseSpace06 = 96.00;
    static const baseSpace07 = 128.00;
    static const baseSpace08 = 160.00;
    static const baseSpace09 = 192.00;
    static const baseSpace10 = 224.00;
}",
    },
  },
  Object {
    "name": "colors.dart",
    "value": Object {
      "content": "
import 'dart:ui';

class SpecifyColor {
    SpecifyColor._();

    static const colorsAccent = Color(0x577CFEFF);
    static const colorsBlack = Color(0x1E212BFF);
    static const colorsGreen = Color(0x58CD52FF);
    static const colorsGrey = Color(0xCCD5E1FF);
    static const colorsOrange = Color(0xFF8E05FF);
    static const colorsRed = Color(0xF5483FFF);
    static const colorsWhite = Color(0xFFFFFFFF);
}",
    },
  },
]
`;

exports[`To Flutter Should generate the color theme with custom file name and custom class name 1`] = `
Array [
  Object {
    "name": "custom-colors-file-name.dart",
    "value": Object {
      "content": "
import 'dart:ui';

class LightTheme {
    LightTheme._();

    static const colorsAccent = Color(0x577CFEFF);
    static const colorsBlack = Color(0x1E212BFF);
    static const colorsGreen = Color(0x58CD52FF);
    static const colorsGrey = Color(0xCCD5E1FF);
    static const colorsOrange = Color(0xFF8E05FF);
    static const colorsRed = Color(0xF5483FFF);
    static const colorsWhite = Color(0xFFFFFFFF);
}",
    },
  },
]
`;

exports[`To Flutter Should generate the measurement theme with custom file name, class name and ratio 1`] = `
Array [
  Object {
    "name": "custom-measurements-file-name.dart",
    "value": Object {
      "content": "
import 'dart:ui';

class CustomTheme {
    CustomTheme._();

    static const baseSpace01 = 12.00;
    static const baseSpace02 = 24.00;
    static const baseSpace03 = 36.00;
    static const baseSpace04 = 48.00;
    static const baseSpace05 = 96.00;
    static const baseSpace06 = 144.00;
    static const baseSpace07 = 192.00;
    static const baseSpace08 = 240.00;
    static const baseSpace09 = 288.00;
    static const baseSpace10 = 336.00;
}",
    },
  },
  Object {
    "name": "colors.dart",
    "value": Object {
      "content": "
import 'dart:ui';

class SpecifyColor {
    SpecifyColor._();

    static const colorsAccent = Color(0x577CFEFF);
    static const colorsBlack = Color(0x1E212BFF);
    static const colorsGreen = Color(0x58CD52FF);
    static const colorsGrey = Color(0xCCD5E1FF);
    static const colorsOrange = Color(0xFF8E05FF);
    static const colorsRed = Color(0xF5483FFF);
    static const colorsWhite = Color(0xFFFFFFFF);
}",
    },
  },
]
`;